src/ScriptEditor/NetscriptDefinitions.d.ts

```diff
@@ -281,6 +281,37 @@ interface RunOptions {
   preventDuplicates?: boolean;
 }
 
+/** @public */
+interface SpawnOptions {
+  /** Number of threads that the script will run with, defaults to 1 */
+  threads?: number;
+  /** Whether this script is excluded from saves, defaults to false */
+  temporary?: boolean;
+  /**
+   * The RAM allocation to launch each thread of the script with.
+   *
+   * Lowering this will <i>not</i> automatically let you get away with using less RAM:
+   * the dynamic RAM check enforces that all {@link NS} functions actually called incur their cost.
+   * However, if you know that certain functions that are statically present (and thus included
+   * in the static RAM cost) will never be called in a particular circumstance, you can use
+   * this to avoid paying for them.
+   *
+   * You can also use this to <i>increase</i> the RAM if the static RAM checker has missed functions
+   * that you need to call.
+   *
+   * Must be greater-or-equal to the base RAM cost. Defaults to the statically calculated cost.
+   */
+  ramOverride?: number;
+  /**
+   * Should we fail to run if another instance is running with the exact same arguments?
+   * This used to be the default behavior, now defaults to false.
+   */
+  preventDuplicates?: boolean;
+  /**
+   * Number of milliseconds to delay before spawning script, defaults to 10000 */
+  spawnDelay?: number;
```

Comment:
Optional: Much like `additionalMs` in HGWOptions, I would call this `spawnDelayMs` to make it clear in the name that it's milliseconds. Time units that aren't strongly typed are *very* easy to get mixed up.

Reply:
Very good :)